frameworks/Swift/vapor/Sources/vapor-tfb-mysql/Routes.swift

```diff
@@ -1,73 +1,73 @@
 import Vapor
-import TfbCommon
 import Foundation
+import TfbCommon
+import FluentMySQL
 
-final class Routes: RouteCollection {
-    func build(_ builder: RouteBuilder) throws {
-        builder.get("json") { req in
-            return try JSON(node: Message("Hello, World!"))
-        }
+struct EmptyJSON: Content {}
 
-        builder.get("plaintext") { req in
-            return "Hello, world!"
-        }
+public func routes(_ router: Router) throws {
 
-        // response to requests to /info domain
-        // with a description of the request
-        builder.get("info") { req in
-            return req.description
-        }
+    // Test type 2: Single database query
+    router.get("db") { req -> Future<Response> in
+        let worldId = WorldMeta.randomId()
+        return try World.find(worldId, on: req)
+            .flatMap(to: Response.self) { world in
+                guard let world = world
+                    else { return try EmptyJSON().encode(for: req) }
 
-        builder.get("description") { req in return req.description }
-
-        // Test type 2: Single database query
-        builder.get("db") { _ in
-            let worldId = WorldMeta.randomId()
-            return try World.find(worldId)?.makeJSON() ?? JSON(node: .null)
+                return try world.encode(for: req)
         }
+    }
 
-        // Test type 3: Multiple database queries
-        builder.get("queries") { req in
-            let queries = queriesParam(for: req)
-            let ids = (1...queries).map({ _ in WorldMeta.randomId() })
-            let worlds = try ids.flatMap { try World.find($0)?.makeJSON() }
-            return JSON(worlds)
-        }
+    // Test type 3: Multiple database queries
+    router.get("queries", String.parameter) { req -> Future<[World]> in
+        let queries = try queriesParam(for: req)
+        let ids = (1...queries).map({ _ in WorldMeta.randomId() })
 
-        // Test type 4: Fortunes
+        return try World.findWith(ids: ids, on: req)
```

Comment:
Is this executing a `SELECT ...  WHERE id IN (...)` statement?  It needs to do a separate query for each id.  Same deal when initially finding the worlds in the updates test, though you can do a single bulk update after doing the separate finds.

Reply:
Yeah, initially it used to make multiple SELECTS but changed it (ofc didn't realize that was the whole point of the benchmark duh)